rootex/framework/components/game/spring_arm_component.h

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include "component.h"
+#include "components/space/transform_component.h"
+#include "common/common.h"
+#include "framework/component.h"
+#include "framework/components/space/transform_component.h"
+#include "framework/components/visual/model/animated_model_component.h"
+#include "framework/components/physics/capsule_collider_component.h"
+#include "core/state_manager.h"
+
+class SpringArmComponent : public Component
+{
+	COMPONENT(SpringArmComponent, Category::Game);
+	DEPENDS_ON(TransformComponent);
+
+	Vector3 m_DesiredLocalPosition;
+
+public:
+	SpringArmComponent(Entity& owner, const JSON::json& data);
+	~SpringArmComponent() = default;
+
+	float m_Lerp;
+
+	virtual JSON::json getJSON() const override;
+	bool setupData() override;
```

Comment:
Typically, try to put inherited stuff at the bottom of the class definition. This keeps the important API functions easier to read because of reduced noise in the header files